lib/jekyll/document.rb

```diff
@@ -265,6 +265,42 @@ def read(opts = {})
       end
     end
 
+    # Add superdirectories of the special_dir to categories.
+    # In the case of es/_posts, 'es' is added as a category.
+    # In the case of _posts/es, 'es' is NOT added as a category.
+    #
+    # Returns nothing.
+    def categories_from_path(special_dir)
+      superdirs = relative_path.sub(%r!#{special_dir}(.*)!, "")
+        .split(File::SEPARATOR)
+        .reject do |c|
+        c.empty? || c == special_dir || c == basename
+      end
+      merge_data!({ "categories" => superdirs }, :source => "file path")
+    end
+
+    def post_read
+      read_post_data
+    end
```

Comment:
@parkr since `:read_post_data` is a `private_method`, what would be the best way to alias to `post_read`? `def_delegator`?